sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -1,65 +1,415 @@
-    sample.setLastUpdated(now);
-    return (Long) currentSession().save(sample);
+  public Collection<Sample> listAll() throws IOException {
+    return getSample();
   }
 
   @Override
-  public void deleteSample(Sample sample) {
-    currentSession().delete(sample);
+  public Collection<Sample> listAllByReceivedDate(long limit) throws IOException {
+    Query query = currentSession().createQuery("from SampleImpl order by receivedDate desc");
+    query.setMaxResults((int) limit);
+    @SuppressWarnings("unchecked")
+    List<Sample> records = query.list();
+    return fetchSqlStore(records);
+  }
+
+  @Override
+  public Collection<String> listAllSampleTypes() throws IOException {
+    return getJdbcTemplate().queryForList("SELECT name FROM SampleType", String.class);
+  }
+
+  @Override
+  public Collection<Sample> listAllWithLimit(long limit) throws IOException {
+    Query query = currentSession().createQuery("from SampleImpl");
+    query.setMaxResults((int) limit);
+    @SuppressWarnings("unchecked")
+    List<Sample> records = query.list();
+    return fetchSqlStore(records);
+  }
+
+  @Override
+  public Collection<Sample> listByAlias(String alias) throws IOException {
+    Query query = currentSession().createQuery("from SampleImpl where alias = :alias");
+    query.setString("alias", alias);
+    @SuppressWarnings("unchecked")
+    List<Sample> records = query.list();
+    return fetchSqlStore(records);
+  }
+
+  @Override
+  public Collection<Sample> listByExperimentId(long experimentId) throws IOException {
+    Query query = currentSession().createQuery("from SampleImpl where experiment.id like :id");
+    query.setLong("id", experimentId);
+    @SuppressWarnings("unchecked")
+    List<Sample> records = query.list();
+    return fetchSqlStore(records);
+  }
+
+  @Override
+  public Collection<Sample> listByProjectId(long projectId) throws IOException {
+    Query query = currentSession().createQuery("from SampleImpl where project.id like :id");
+    query.setLong("id", projectId);
+    @SuppressWarnings("unchecked")
+    List<Sample> records = query.list();
+    return fetchSqlStore(records);
+  }
+
+  @Override
+  public Collection<Sample> listBySearch(String querystr) throws IOException {
+    Query query = currentSession().createQuery(
+        "from SampleImpl where identificationBarcode like :query or name LIKE :query or alias like :query or description like :query or scientificName like :query");
+    query.setString("query", querystr);
+    @SuppressWarnings("unchecked")
+    List<Sample> records = query.list();
+    return fetchSqlStore(records);
+  }
 
+  @Override
+  public Collection<Sample> listBySubmissionId(long submissionId) throws IOException {
+    Query query = currentSession().createQuery("from SampleImpl where submissionId like :id");
+    query.setLong("id", submissionId);
+    @SuppressWarnings("unchecked")
+    List<Sample> records = query.list();
+    return fetchSqlStore(records);
+  }
```

Comment:
submissionId comes from Submission_Sample table. The original query is:
`public static final String SAMPLES_BY_RELATED_SUBMISSION = SAMPLES_SELECT  + " WHERE sampleId IN (SELECT samples_sampleId FROM Submission_Sample WHERE submission_submissionId=?)"; `